File: Assets/zfoocs/Jprotobuf/JProtobufHelloResponse.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class JProtobufHelloResponse
    {
        public string message;

        public static JProtobufHelloResponse ValueOf(string message)
        {
            var packet = new JProtobufHelloResponse();
            packet.message = message;
            return packet;
        }
    }


    public class JProtobufHelloResponseRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1501;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            JProtobufHelloResponse message = (JProtobufHelloResponse) packet;
            buffer.WriteInt(-1);
            buffer.WriteString(message.message);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            JProtobufHelloResponse packet = new JProtobufHelloResponse();
            string result0 = buffer.ReadString();
            packet.message = result0;
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}

File: Assets/zfoolua/jprotobuf/JProtobufHelloResponse.lua

local JProtobufHelloResponse = {}

function JProtobufHelloResponse:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function JProtobufHelloResponse:protocolId()
    return 1501
end

function JProtobufHelloResponse:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function JProtobufHelloResponse:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = JProtobufHelloResponse:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return JProtobufHelloResponse
